lib/elixir/src/elixir_rewrite.erl

```diff
@@ -173,9 +176,16 @@ inline(_, _, _) -> false.
 
 %% Complex rewrite rules
 
-rewrite(?string_chars, _DotMeta, 'to_string', _Meta, [String]) when is_binary(String) ->
+rewrite(?access, _DotMeta, 'get', _Meta, [nil, Arg], _Env)
+when ?is_literal(Arg) orelse (is_atom(element(1, Arg)) andalso element(3, Arg) == nil) ->
+  nil;
+rewrite(?access, _DotMeta, 'get', Meta, [Arg, _], Env)
+when ?is_literal(Arg) orelse element(1, Arg) == '{}' orelse element(1, Arg) == '<<>>' ->
+  elixir_errors:compile_error(Meta, ?m(Env, file),
+    "the Access is not available for the value: ~ts", ['Elixir.Macro':to_string(Arg)]);
```

Comment:
But this place handles not only Access syntax `[]` but also actual Access calls.